src/SmartTransactionsController.ts

```diff
@@ -39,14 +39,14 @@ import { CHAIN_IDS } from './constants';
 const SECOND = 1000;
 export const DEFAULT_INTERVAL = SECOND * 5;
 
-export interface SmartTransactionsControllerConfig extends BaseConfig {
+export type SmartTransactionsControllerConfig = {
   interval: number;
   clientId: string;
   chainId: string;
   supportedChainIds: string[];
-}
+} & BaseConfig;
```

Comment:
I don't know why our ESLint rules autoconvert an `interface` into a `type this way, but shouldn't this be reversed?

``` suggestion
export type SmartTransactionsControllerConfig = BaseConfig & {
  interval: number;
  clientId: string;
  chainId: string;
  supportedChainIds: string[];
}
```

Reply:
Agreed, we should definitely fix this at the lint level if we can! It looks like the relevant part of the TypeScript ESLint rule is here: https://github.com/typescript-eslint/typescript-eslint/blob/8951732e839a94d7c436d5774d473e6a05e8a9f1/packages/eslint-plugin/src/rules/consistent-type-definitions.ts#L105. So we could either fork this rule or submit a PR. I'll see if I can look into this soon.